src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ConnectHelper.cs

```diff
@@ -125,6 +125,8 @@ private static async ValueTask<SslStream> EstablishSslConnectionAsyncCore(bool a
                     throw CancellationHelper.CreateOperationCanceledException(e, cancellationToken);
                 }
 
+                // ToDo: let this exception buuble up to the caller, or should we somehow recognize that server didn't allow
+                //       particular HTTP version in ALPN (how? Furtik will know) and use NSE instead.
```

Comment:
What's the plan for this TODO?  Should it be deleted? Is there an issue tracking it?

Reply:
We decided to leave it as-is now. I must have forgotten to delete the comment :disappointed: 
There's also a resolved discussion on this topic, but I assume you've already seen it.